lib/kubernetes-deploy/kubernetes_resource/service.rb

```diff
@@ -11,16 +11,19 @@ def sync
     end
 
     def status
-      if @num_pods_selected.blank?
+      if !requires_endpoints?
+        "Doesn't require any endpoint"
+      elsif @num_pods_selected.blank?
         "Failed to count related pods"
       elsif selects_some_pods?
         "Selects at least 1 pod"
       else
-        "Selects 0 pods"
+        "Unknown"
```

Comment:
The elsif condition selects_some_pods? applies to when pods are found. 
The else condition should cover the case when no pods are found for the service selector while replica is > 0. Besides that, are there are cases that will fall into the else clause? Message "unknown" is not clear to me.

Reply:
Hm, it seems like we'll only arrive here if we expected to select some pods but saw <= 0. If that's true, `Selects 0 pods` might still be a better message, but otoh I don't hate `Unknown` since this is a catch-all fallback case. I think you have a point, but I wouldn't block merging the PR on this if the author feels differently.

Reply:
I replaced the message with `Unknown` to be consistent with the other services, but I did realized we could have this message for good reasons.